chia/harvester/harvester.py

```diff
@@ -3,65 +3,58 @@
 import logging
 from concurrent.futures.thread import ThreadPoolExecutor
 from pathlib import Path
-from typing import Callable, Dict, List, Optional, Set, Tuple
-
-from blspy import G1Element
+from typing import Callable, Dict, List, Optional, Tuple
 
 import chia.server.ws_connection as ws  # lgtm [py/import-and-import-from]
 from chia.consensus.constants import ConsensusConstants
-from chia.plotting.plot_tools import PlotInfo
-from chia.plotting.plot_tools import add_plot_directory as add_plot_directory_pt
-from chia.plotting.plot_tools import get_plot_directories as get_plot_directories_pt
-from chia.plotting.plot_tools import load_plots
-from chia.plotting.plot_tools import remove_plot_directory as remove_plot_directory_pt
+from chia.plotting.plot_tools import PlotsRefreshParameter, PlotManager, PlotRefreshResult
 
 log = logging.getLogger(__name__)
 
 
 class Harvester:
-    provers: Dict[Path, PlotInfo]
-    failed_to_open_filenames: Dict[Path, int]
-    no_key_filenames: Set[Path]
-    farmer_public_keys: List[G1Element]
-    pool_public_keys: List[G1Element]
+    plot_manager: PlotManager
     root_path: Path
     _is_shutdown: bool
     executor: ThreadPoolExecutor
     state_changed_callback: Optional[Callable]
     cached_challenges: List
     constants: ConsensusConstants
     _refresh_lock: asyncio.Lock
+    event_loop: asyncio.events.AbstractEventLoop
 
     def __init__(self, root_path: Path, config: Dict, constants: ConsensusConstants):
-        self.root_path = root_path
-
-        # From filename to prover
-        self.provers = {}
-        self.failed_to_open_filenames = {}
-        self.no_key_filenames = set()
+        self.log = log
+        # TODO, remove checks below later after some versions / time
+        refresh_parameter: PlotsRefreshParameter = PlotsRefreshParameter()
+        if "plot_loading_frequency_seconds" in config:
+            self.log.warning(
+                "plot_loading_frequency_seconds is deprecated but found in config. Replace it with the "
```

Comment:
i would consider removing this, your setting default params anyway and i think this might scare some new farmers, maybe change from warning to debug

Reply:
Agree its probably better to not print such a warning, i pushed four more commits and added a migration step to the `load_config` in 5947039889bf16026d6fea9ed0b6af93209c98d3.

Reply:
cool but i think we should raise the config migration in the meeting, its useful but could be bad if we mess up users setup, what we did until now is set a hardcoded default for new params

Reply:
> cool but i think we should raise the config migration in the meeting

Sure we can talk about it later in the meeting :+1:

> but could be bad if we mess up users setup

How do you think it could be messed?

Reply:
i suggested the exact same thing few months back and we had the same discussion

Reply:
we can mess up the users setup if we corrupt the config file/save a bad value in there

Reply:
I dropped that migration stuff